rust/lance-linalg/src/kmeans.rs

```diff
@@ -146,6 +170,20 @@ where
     metric_type: MetricType,
 }
 
+fn split_clusters<T: Float + DivAssign>(cnts: &mut [u64], centroids: &mut [T], dimension: usize) {
+    for i in 0..cnts.len() {
+        if cnts[i] == 0 {
+            let largest_idx = argmax(cnts.iter().copied()).unwrap() as usize;
+            cnts[i] = cnts[largest_idx] / 2;
+            cnts[largest_idx] /= 2;
```

Comment:
How does this handle odd counts? Is there any requirement for these counts to be exact?

Reply:
At this moment, it does not matter. The new centroids ( `sum / cnt` ) has already been computed. And here, the cnts are just useful to find the empty / largest clusters.